runtime/lib/capi/ExecutionContext.hpp

```diff
@@ -259,6 +260,9 @@ class RTDevice {
 
     void setDeviceStatus(RTDeviceStatus new_status) noexcept { status = new_status; }
 
+    void setDeviceAutoQubitManagementMode(bool mode) { auto_qubit_management = mode; }
+    bool getDeviceAutoQubitManagementMode() { return auto_qubit_management; }
```

Comment:
Function name is a bit convoluted
```suggestion
    void setQubitManagementMode(bool mode) { auto_qubit_management = mode; }
    bool getQubitManagementMode() { return auto_qubit_management; }
```

Reply:
Although tbh, I don't really see a reason to update the mode, so using the constructor for setting seems better? 
```suggestion
    bool getQubitManagementMode() { return auto_qubit_management; }
```

Reply:
https://github.com/PennyLaneAI/catalyst/pull/1788/commits/c159c440672b2bcfb5930c10d1b9c47872d17f1f moved to constructor